mysql/provider.go

```diff
@@ -175,6 +213,46 @@ func providerConfigure(ctx context.Context, d *schema.ResourceData) (interface{}
 	var allowNativePasswords = authPlugin == nativePasswords
 	var password = d.Get("password").(string)
 	var iam_auth = d.Get("iam_database_authentication").(bool)
+	var tlsConfig = d.Get("tls").(string)
+	var tlsConfigStruct *tls.Config
+	var customTLS CustomTLS
```

Comment:
Could you please move this line as close to the usage as possible? It's not necessary to declare all variable here. For getting other configs, maybe it's more readable when one doesn't have to go thru whole function when changing parameter name. But that's it.

Reply:
I think there isn't much we can do about it, considering their scopes. The `if` block a couple of lines below have assignments to them, and they are used outside of the scope of the `if` block later in the function. I'll move `customTLS` to be declared inside of the block, though